java/client/test/org/openqa/selenium/support/pagefactory/ByChainedTest.java

```diff
@@ -247,6 +253,47 @@ public void findElementsTwoByEmptyChild() {
     assertThat(by.findElements(driver)).isEqualTo(elems5);
   }
 
+  @Test
+  public void findElementsThreeBy_firstFindsOne_secondEmpty() {
+    final AllDriver driver = mock(AllDriver.class);
+    final WebElement elem1 = mock(WebElement.class, "webElement1");
+
+    String by1Name = "by1";
+    By by1 = By.name(by1Name);
+    By by2 = By.name("by2");
+    By by3 = By.name("by3");
+
+    when(driver.findElementsByName(by1Name)).thenReturn(asList(elem1));
+    when(elem1.findElements(by2)).thenReturn(NO_ELEMENTS);
+
+    ByChained by = new ByChained(by1, by2, by3);
+
+    assertThat(by.findElements(driver)).isEmpty();
+    verify(elem1, never()).findElements(by3);
+  }
+
+  @Test
+  public void findElementThreeBy_firstFindsTwo_secondEmpty() {
+    final AllDriver driver = mock(AllDriver.class);
+    final WebElement elem1 = mock(WebElement.class, "webElement1");
+    final WebElement elem2 = mock(WebElement.class, "webElement2");
+
+    String by1Name = "by1";
+    By by1 = By.name(by1Name);
+    By by2 = By.name("by2");
+    By by3 = By.name("by3");
+
+    when(driver.findElementsByName(by1Name)).thenReturn(asList(elem1, elem2));
+    when(elem1.findElements(by2)).thenReturn(NO_ELEMENTS);
+    when(elem2.findElements(by2)).thenReturn(NO_ELEMENTS);
+
+    ByChained by = new ByChained(by1, by2, by3);
+
+    assertThat(by.findElements(driver)).isEmpty();
+    verify(elem1, never()).findElements(by3);
+    verify(elem2, never()).findElements(by3);
+  }
+
```

Comment:
Is it just me or are these two tests not compiling?

`findElementsByName` does not exist and probably some `mockito` import is missing?

Reply:
Sorry, that's possible.

The build is not working on my IDEA, and I don't know how to run the tests from the command line: `bazel test ...??`.

The documentation about building is scattered and outdated, it seems to me:
- some pieces are on the [README](https://github.com/SeleniumHQ/selenium/blob/trunk/README.md#common-build-targets)
- some on [this wiki page](https://github.com/SeleniumHQ/selenium/wiki/Build-Instructions), linked [from the README](https://github.com/SeleniumHQ/selenium/blob/trunk/README.md#useful-resources)
- in [Build Instructions](https://github.com/SeleniumHQ/selenium/wiki/Build-Instructions), some commands like `bazel test java/client/test/org/openqa/selenium/chrome` do not work anymore